gfx.cpp

```diff
@@ -967,8 +968,28 @@ class GfxInternal
         window_height_ = window_rect.bottom - window_rect.top;
 
         IDXGIOutput *output = nullptr;
-        DXGI_COLOR_SPACE_TYPE color_space = DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709;
-        if(SUCCEEDED(adapter_->EnumOutputs(0, &output)))
+        color_space_ = DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709;
+        UINT output_i  = 0;
+        LONG best_area = -1;
+        IDXGIOutput *current_output;
+        while(adapter_->EnumOutputs(output_i, &current_output) != DXGI_ERROR_NOT_FOUND)
+        {
+            DXGI_OUTPUT_DESC output_desc;
+            if(SUCCEEDED(current_output->GetDesc(&output_desc))) {
+                RECT rect = output_desc.DesktopCoordinates;
+                int intersect_area =
+                    GFX_MAX(0L, GFX_MIN(window_rect.right, rect.right) - GFX_MAX(window_rect.left, rect.left)) *
+                    GFX_MAX(0l, GFX_MIN(window_rect.bottom, rect.bottom) - GFX_MAX(window_rect.top, rect.top));
+                if(intersect_area > best_area)
+                {
+                    output    = current_output;
+                    best_area = intersect_area;
+                }
+            }
+            if(current_output != output) current_output->Release();
+            output_i++;
+        }
+        if(output != nullptr)
```

Comment:
Improved display detection when user has multiple monitors and the window is spanning multiple displays